### PR TITLE
feat: send connection repository work - part 2 (AR-1483)

### PR DESCRIPTION
**Pull request by yamilmedina**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [X] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [X] contains a reference JIRA issue number like `SQPIT-764`

- The **PR Description**
  - [X] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

https://wearezeta.atlassian.net/browse/AR-1483

### Issues

This work it's related to the creation of a connection of a user, "Send a connection request" in detail this PR takes care of the repo layer, namely: Calling the Network layer and persisting locally the response.

You could see prev. PR implemented here -> #468 

#### Test Coverage (Optional)

- [X] I have added automated test to this contribution

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [X] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [X] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: logic/src/commonTest/kotlin/com/wire/kalium/logic/data/connection/ConnectionRepositoryTest.kt [hunk 1]
```diff
@@ -72,6 +74,34 @@ class ConnectionRepositoryTest {
...
+            .suspendFunction(connectionApi::createConnection)
+            .whenInvokedWith(eq(userId))
+            .then { NetworkResponse.Success(connection1, mapOf(), 200) }
+        given(conversationDAO)
+            .suspendFunction(conversationDAO::insertOrUpdateOneOnOneMemberWithConnectionStatus)
+            .whenInvokedWith(eq(UserIDEntity(userId.value, userId.domain)), any(), any())
+            .then { _, _, _ -> return@then }
+
+        // when
+        val result = connectionRepository.sendUserConnection(com.wire.kalium.logic.data.user.UserId(userId.value, userId.domain))
```
Comment (alexandreferris) [inline logic/src/commonTest/kotlin/com/wire/kalium/logic/data/connection/ConnectionRepositoryTest.kt:91]: maybe we could add `com.wire.kalium.logic.data.user.UserId` as a proper import or with a different name in case there is already one with the same name 🤔

**Review by alexandreferris — CHANGES_REQUESTED** [review]
Great work @yamilmedina !

Just got one change about the import on tests

**Review by gongracr — COMMENTED** [review]
Would it be possible to add a few more tests when the api or the storage call fails? This is usually where most of edge cases will pile up so imho these tests are the important ones. 

If you want to reuse some error test setups I suggest you to check the tests on `GetMessageAssetUseCaseTest` and `AssetImageFetcherTest` for inspiration 😛

File: logic/src/commonTest/kotlin/com/wire/kalium/logic/data/connection/ConnectionRepositoryTest.kt (hunk 1 above)
Comment (yamilmedina) [reply logic/src/commonTest/kotlin/com/wire/kalium/logic/data/connection/ConnectionRepositoryTest.kt]: Good idea, I did some changes to have it imported with an alias

**Review by alexandreferris — APPROVED** [review]
🚀
